### PR TITLE
507 add Path.hasSameTextualContentAs/hasSameBinaryContent to infix API 

### PR DESCRIPTION
**Pull request by z13z**
______________________________________
I confirm that I have read the [Contributor Agreements v1.0](https://github.com/robstoll/atrium/blob/master/.github/Contributor%20Agreements%20v1.0.txt), agree to be bound on them and confirm that my contribution is compliant.

### CODE REVIEW

File: apis/fluent-en_GB/atrium-api-fluent-en_GB-jvm/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/PathAssertionsSpec.kt [hunk 1]
```diff
@@ -20,8 +21,15 @@ class PathAssertionsSpec : ch.tutteli.atrium.specs.integration.PathAssertionsSpe
     fun0(Expect<Path>::isRegularFile),
     fun0(Expect<Path>::isDirectory),
     fun1(Expect<Path>::hasSameBinaryContentAs),
-    fun3(Expect<Path>::hasSameTextualContentAs)
+    fun3(Expect<Path>::hasSameTextualContentAs),
+    fun1(Companion::hasSameTextualContentAsDefaultArgs)
 ) {
+
+    companion object : WithAsciiReporter() {
```
Comment (robstoll) [inline apis/fluent-en_GB/atrium-api-fluent-en_GB-jvm/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/PathAssertionsSpec.kt:28]:
```suggestion
    companion object {
```

File: apis/infix-en_GB/atrium-api-infix-en_GB-jvm/src/main/kotlin/ch/tutteli/atrium/api/infix/en_GB/creating/encoding/EncodingWithCreator.kt [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+/**
+ *  Parameter object which combines [path] (as [Path]), [sourceCharset] (as [Charset]), [targetCharset] (as [Charset])
+ *  with an [assertionCreator] which defines assertions for a resulting feature of type [E].
+ *
+ *  Use the function `withEncoding(Path, Charset, Charset) { ... }` to create this representation.
+ *
+ *  @since 0.13.0
+ */
+data class EncodingWithCreator<E> internal constructor(val path: Path, val sourceCharset: Charset, val targetCharset: Charset,
+                                                       val assertionCreator: Expect<E>)
```
Comment (robstoll) [inline apis/infix-en_GB/atrium-api-infix-en_GB-jvm/src/main/kotlin/ch/tutteli/atrium/api/infix/en_GB/creating/encoding/EncodingWithCreator.kt:16]:
```suggestion
data class EncodingWithCreator<E> internal constructor(val path: Path, val sourceCharset: Charset, val targetCharset: Charset)
```

File: apis/infix-en_GB/atrium-api-infix-en_GB-jvm/src/main/kotlin/ch/tutteli/atrium/api/infix/en_GB/pathAssertions.kt [hunk 3]
```diff
@@ -298,3 +300,55 @@ val <T : Path> Expect<T>.extension: Expect<String>
...
+ * as [targetPath].
+ *
+ * @return An [Expect] for the current subject of the assertion.
+ * @throws AssertionError Might throw an [AssertionError] if the assertion made is not correct.
+ *
+ * @since 0.13.0
+ */
+infix fun <T : Path> Expect<T>.hasSameTextualContentAs(
+    targetPath: Path
+): Expect<T> = hasSameTextualContentAs(withEncoding(targetPath, this))
```
Comment (robstoll) [inline apis/infix-en_GB/atrium-api-infix-en_GB-jvm/src/main/kotlin/ch/tutteli/atrium/api/infix/en_GB/pathAssertions.kt:315]:
```suggestion
): Expect<T> = hasSameTextualContentAs(withEncoding(targetPath))
```

File: apis/infix-en_GB/atrium-api-infix-en_GB-jvm/src/main/kotlin/ch/tutteli/atrium/api/infix/en_GB/pathAssertions.kt [hunk 4]
```diff
@@ -298,3 +300,55 @@ val <T : Path> Expect<T>.extension: Expect<String>
...
+ * @throws AssertionError Might throw an [AssertionError] if the assertion made is not correct.
+ *
+ * @since 0.13.0
+ */
+infix fun <T : Path> Expect<T>.hasSameTextualContentAs(encodingWithCreator: EncodingWithCreator<T>):
+    Expect<T> = _logicAppend {
+        hasSameTextualContentAs(
+            encodingWithCreator.path, encodingWithCreator.sourceCharset, encodingWithCreator.targetCharset
+        )
+    }
```
Comment (robstoll) [inline apis/infix-en_GB/atrium-api-infix-en_GB-jvm/src/main/kotlin/ch/tutteli/atrium/api/infix/en_GB/pathAssertions.kt:333]:
```suggestion
infix fun <T : Path> Expect<T>.hasSameTextualContentAs(encodingWithCreator: EncodingWithCreator<T>): Expect<T> =
    _logicAppend {
        hasSameTextualContentAs(
            encodingWithCreator.path, encodingWithCreator.sourceCharset, encodingWithCreator.targetCharset
        )
    }
```

File: apis/infix-en_GB/atrium-api-infix-en_GB-jvm/src/main/kotlin/ch/tutteli/atrium/api/infix/en_GB/pathAssertions.kt [hunk 5]
```diff
@@ -298,3 +300,55 @@ val <T : Path> Expect<T>.extension: Expect<String>
...
+ *
+ * @return An [Expect] for the current subject of the assertion.
+ * @throws AssertionError Might throw an [AssertionError] if the assertion made is not correct.
+ *
+ * @since 0.13.0
+ */
+infix fun <T : Path> Expect<T>.hasSameBinaryContentAs(targetPath: Path):
+    Expect<T> = _logicAppend {
+    hasSameBinaryContentAs(targetPath)
+}
```
Comment (robstoll) [inline apis/infix-en_GB/atrium-api-infix-en_GB-jvm/src/main/kotlin/ch/tutteli/atrium/api/infix/en_GB/pathAssertions.kt:346]: please don't put the return type on a new line
```suggestion
infix fun <T : Path> Expect<T>.hasSameBinaryContentAs(targetPath: Path): Expect<T> = 
    _logicAppend { hasSameBinaryContentAs(targetPath) }
```

File: apis/infix-en_GB/atrium-api-infix-en_GB-jvm/src/main/kotlin/ch/tutteli/atrium/api/infix/en_GB/pathAssertions.kt [hunk 6]
```diff
@@ -298,3 +300,55 @@ val <T : Path> Expect<T>.extension: Expect<String>
...
+
+
+/**
+ * Helper function to create an [EncodingWithCreator] based on the given [path] and [assertionCreator].
+ */
+fun <T : Path> withEncoding(path: Path, assertionCreator: Expect<T>,
+                            sourceCharset: Charset = Charsets.UTF_8, targetCharset: Charset = Charsets.UTF_8):
+    EncodingWithCreator<T> = EncodingWithCreator(path, sourceCharset, targetCharset, assertionCreator)
```
Comment (robstoll) [inline apis/infix-en_GB/atrium-api-infix-en_GB-jvm/src/main/kotlin/ch/tutteli/atrium/api/infix/en_GB/pathAssertions.kt:354]:
```suggestion
fun <T : Path> withEncoding(path: Path
                            sourceCharset: Charset = Charsets.UTF_8, targetCharset: Charset = Charsets.UTF_8):
    EncodingWithCreator<T> = EncodingWithCreator(path, sourceCharset, targetCharset)
```

File: misc/deprecated/apis/fluent-en_GB-jdk8/atrium-api-fluent-en_GB-jdk8-jvm/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/jdk8/PathAssertionsSpec.kt [hunk 7]
```diff
@@ -23,8 +24,15 @@ class PathAssertionsSpec : ch.tutteli.atrium.specs.integration.PathAssertionsSpe
     fun0(Expect<Path>::isRegularFile),
     fun0(Expect<Path>::isDirectory),
     fun1(Expect<Path>::hasSameBinaryContentAs),
-    fun3(Expect<Path>::hasSameTextualContentAs)
+    fun3(Expect<Path>::hasSameTextualContentAs),
+    fun1(Companion::hasSameTextualContentAsDefaultArgs)
 ) {
+
+    companion object : WithAsciiReporter() {
```
Comment (robstoll) [inline misc/deprecated/apis/fluent-en_GB-jdk8/atrium-api-fluent-en_GB-jdk8-jvm/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/jdk8/PathAssertionsSpec.kt:31]:
```suggestion
    companion object {
```

**Comment by robstoll** [comment]
As side notice, you still have a superfluous commit in your history but don't worry, as you don't have conflicts, I can squash your PR in the end. Try to avoid it in your next PR. Always start from the current upstream/master

**Comment by robstoll** [comment]
@z13z the main part is missing in this PR, the API changes. Do you plan to adjust this PR?

**Comment by z13z** [comment]
@robstoll 
Can you tell me changes are needed for this pull request, please

**Comment by robstoll** [comment]
@z13z see review above

**Pull request merged**